### PR TITLE
added validation to models

### PR DESCRIPTION
**Pull request by jdexcond**
**summary**
so i added the db validation to the models and like it works but my issue is i couldn't figure out how to get it to print the error messages. so like for example, in the list model, i have `validates_uniqueness_of :item, case_sensitive: false, message: "item already exists in your list` so that if someone already has butter in their list, this would prevent them from adding butter again. and it works, it won't add butter twice but i couldn't get any error to pop up on the screen that prints out the "item already exists in your list" message. actually now that im thinking about this, i feel like if even someone else has butter on their list, it would stop u from putting butter in your list because this validation is preventing the db from having 2 butters. ok gonna draft this for now and request the help from my peers tmrw.

### CODE REVIEW

**Comment by e33lin** [comment]
hello peer
i think we're going to have to forego that uniqueness line bc i agree, no one else will be able to add butter. i think the rest of the validations are good enough from a database level (since we dont actually have a distinguished foreign key for linking the tables together or just generally a legit database system). i think its sufficient to check for that kind of uniqueness at the controller level and throw error there before adding

File: smart-cart/app/controllers/lists_controller.rb [hunk 1]
```diff
@@ -41,7 +41,7 @@ def create
     #   end
     # end
 
-    List.create(list_id: $session_id, item: params[:item], quantity: 1)
+    @list = List.create(list_id: $session_id, item: params[:item], quantity: 1)
```
Comment (e33lin) [inline smart-cart/app/controllers/lists_controller.rb:44]: wait whats the difference in doing this not that i have an opinion just for my own knowledge LOL

File: smart-cart/app/controllers/lists_controller.rb [hunk 2]
```diff
@@ -41,7 +41,7 @@ def create
...
```
Comment (e33lin) [inline smart-cart/app/controllers/lists_controller.rb:45]: maybe we could add like:

```suggestion
if (!List.find_by(list_id: $session_id, item: params[:item]).empty?)
    throw some sort of flash error or whatever it is
end
```

**Comment by jdexcond** [comment]
idk why this conflicts thing doesn't make sense to me but like i can't see the conflicts so idk how to fix this

**Review by e33lin — APPROVED** [review]
fixed conflicts

**Pull request merged**